WordPress/src/main/java/org/wordpress/android/ui/uploads/AbstractUploadHandler.java

```diff
@@ -0,0 +1,7 @@
+package org.wordpress.android.ui.uploads;
+
+abstract class AbstractUploadHandler {
+    abstract void unregister();
+    abstract boolean hasInProgressUploads();
+    abstract void cancelInProgressUploads();
+}
```

Comment:
I'm not a fan of 1, because you don't add an abstract `register()` without exposing the method to `UploadService`. And I'd prefer not to use a different registration pattern than every other service/activity/fragment in the app.

I'm okay with 2. I used an abstract class instead because I expected to have some common implementation I would add there, but that didn't end up happening. Logically two interfaces would probably make more sense than 1, an `UploadHandler` with `hasInProgressUploads()` and `cancelInProgressUploads()`, and an `Unregisterable` (probably there can be a better name that that) with `unregister()`. But I think that's starting to just become overly fancy and doesn't actually add anything.

I just wanted to create the suggestion that future methods used by both handlers should use the same naming, so it's easier to follow what the service/handlers are doing. I think 2 still accomplishes that, so I'll go with that. We can go back to an abstract class later if we find it useful to have some common implementation after all.